src/test/scala/org/hammerlab/magic/rdd/LazyZippedWithIndexRDDTest.scala

```diff
@@ -0,0 +1,13 @@
+package org.hammerlab.magic.rdd
+
+import org.hammerlab.magic.rdd.LazyZippedWithIndexRDD._
+import org.hammerlab.magic.util.SparkSuite
+
+class LazyZippedWithIndexRDDTest extends SparkSuite {
+  test("match with zipWithIndex") {
+    val rdd = sc.parallelize(0 until 10).filter(_ % 2 == 0)
+    val res1 = rdd.lazyZipWithIndex.collect
+    val res2 = rdd.zipWithIndex.collect
```

Comment:
would also be nice to have a `TestSparkListener` check here; let me send a separate PR with that code, though that doesn't need to block merging this PR if you respond to my other comment above